### DISCUSSION TITLE
A question on running llama on neorv32

### DISCUSSION THREAD

**Discussion by mmcheraghi** [stnolting/neorv32#995] [General]
Hi
I'm thrilled to run LLaMA.cpp on neorv32 processor. Do you think that it's possible to do that?

**Comment by Sam-Vervaeck** [comment]
Hi! Can you provide more context than just a file name? I recently tried running a cpp library on the NEORV32. The cpp language should not be a problem (documentation for cpp is just a bit barebones at the moment). However, you should check first if the LLAMA library is made for 32-bit systems. My cpp library was not and that was the actual source of some implementation problems.

**Reply by mmcheraghi** [reply]
Hi. Sorry you're right. LLaMA.cpp is a project that implements large language models using cpp language. 
https://github.com/ggerganov/llama.cpp
I ran it using `Termux` app on my Android phone, I think such a project needs OS, is it right?

**Reply by Sam-Vervaeck** [reply]
Looking at that project readme the setup mentions UNIX-like and Windows systems to set it up, so it def seems like it needs an OS to run on. Running it bare-metal on NEORV32 seems like a nono. You could maybe try to get it to run on the Zephyr OS port since that is close to Linux but that would probably require quite a bit of work.

**Reply by mmcheraghi** [reply]
Actually I couldn't find any OS specification on readme, thanks.

**Reply by Sam-Vervaeck** [reply]
It is not explicit, but they mention windows, unix, and docker. That all points to an OS.

**Reply by stnolting** [reply]
If it is just "plain C(++)" then it "should" be no "problem"... 😅

I have only briefly skimmed the documentation of llama and so far I could not identify any requirements beyond "plain C++" code... However, things will get complicated if llama requires some kind of advanced runtime environment like virtual memory, forking and threading, OS calls, ...

**Reply by mmcheraghi** [reply]
Thanks a lot for your response.
